src/downloader.rs

```
@@ -531,8 +532,12 @@ impl DownloaderInternal {
 			}
 		);
 		let path = Path::new(&path).to_owned();
-		let path_clone = path.clone();
+		if !config.overwrite && path.exists() {
+			// track already downloaded
+			return Err(SpotifyError::AlreadyDownloaded);
```

Should this really return an error? Maybe we should silently skip the download.

---

this is how i implemented it in the first place but it required some function signature changes to propagate it to the top which looked a little more ugly in the code, cause the function returned also a bool values which was called `downloaded`.

if you know a better way on how to silently return, it would be perfect :)